pulsar-metadata/src/main/java/org/apache/pulsar/metadata/impl/AbstractMetadataStore.java

```diff
@@ -207,7 +207,7 @@ public void accept(Notification n) {
 
     @Override
     public void close() throws Exception {
-        executor.shutdownNow();
+        executor.shutdown();
```

Comment:
Why do we need this change? The `shutdownNow()` will ensure that the thread is interrupted if it was waiting on anything. I think that, as a general rule, it should be preferable to wait for the thread to exit on its own.

Reply:
Thanks for the review @merlimat .

> Why do we need this change?

Valid point. It's not actually needed. 

When I was debugging the flakiness, the key issue of the flakiness of MessageIdTest was related to the restarting of the broker not happening "cleanly". I started making an experiment where I'd revisit the shutdown of the broker. However these changes weren't relevant for making the test stable. I'll remove them completely out of this PR.

The relevant problem in shutdown was in BrokerService where `listenChannel.close()` doesn't use `.await` to synchronously wait for the completion of shutting down the listeners. I'll submit that fix as a separate PR. @sijie suggested in his review comments to split some changes as separate PRs.

Reply:
I have now removed the executor shutdown/shutdownNow related changes from this PR. PTAL @merlimat